app/models/procedure.rb

```diff
@@ -944,11 +944,11 @@ def pieces_jointes_list?
   end
 
   def pieces_jointes_list_without_conditionnal
-    active_revision.types_de_champ_public.not_condition.filter(&:piece_justificative?)
+    pieces_jointes_list(active_revision.revision_types_de_champ_public.includes(:type_de_champ).where(types_de_champ: { condition: nil }))
```

Comment:
on peut passer par `active_revision.revision_types_de_champ` pour avoir tous les tdc sans s'occuper de savoir s'ils sont dans des blocs répétables ou pas.

```suggestion
    active_revision.revision_types_de_champ.public_only.where(types_de_champ: { condition: nil, type_champ: 'piece_justificative' }))
```

Reply:
Oui, mais du coup, on n'a pas la garantie de l'ordre pour l'affichage. J'ai l'impression que préserver l'ordre du formulaire est une bonne idée pour l'UX.

Reply:
J'ai pris le meilleur de chacune de vos 2 suggestions: effectivement on a besoin de plus de finesse pour l'ordre, mais aussi pour afficher le nom du bloc répétable